### PR TITLE
Add admin interface to enforce 2FA

### PR DESCRIPTION
**Pull request by ChristophWurst**
Ref https://github.com/nextcloud/server/issues/11017.

---

## Screenshot

bildschirmfoto von 2018-10-08 14-00-21

cc @nextcloud/designers for the design

### CODE REVIEW

File: settings/js/0.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
...ED:s,VIEW:u}},function(t,e,n){window,t.exports=function(t){var e={};function n(r){if(e[r])return e[r].exports;var i=e[r]={i:r,l:!1,exports:{}};return t[r].call(i.exports,i,i.exports,n),i.l=!0,i.exports}return n.m=t,n.c=e,n.d=function(t,e,r){n.o(t,e)||Object.defineProperty(t,e,{configurable:!1,enumerable:!0,get:r})},n.r=function(t){Object.defineProperty(t,"__esModule",{value:!0})},n.n=function(t){var e=t&&t.__esModule?function(){return t.default}:function(){return t};return n.d(e,"a",e),e},n.o=function(t,e){return Object.prototype.hasOwnProperty.call(t,e)},n.p="",n(n.s=3)}([function(t,e,n){var r;!function(i){"use strict";var o={},a=/d{1,4}|M{1,4}|YY(?:YY)?|S{1,3}|Do|ZZ|([HhMsDm])\1?|[aA]|"[^"]*"|'[^']*'/g,s=/\d\d?/,u=/[0-9]*['a-z\u00A0-\u05FF\u0700-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+|[\u0600-\u06FF\/]+(\s*?[\u0600-\u06FF]+){1,2}/i,c=/\[([^]*?)\]/gm,l=function(){};function f(t,e){for(var n=[],r=0,i=t.length;r<i;r++)n.push(t[r].substr(0,e));return n}function p(t){return function(e,n,r){var i=r[t].indexOf(n.charAt(0).toUpperCase()+n.substr(1).toLowerCase());~i&&(e.month=i)}}function h(t,e){for(t=String(t),e=e||2;t.length<e;)t="0"+t;return t}var d=["Sunday","Monday","Tuesday","Wednesday"
```
Comment (MorrisJobke) [inline settings/js/0.js]: Is there a reason why it was renamed like this?

Comment (rullzer) [reply settings/js/0.js]: I guess no reason. Just git trying to be smart.

Comment (ChristophWurst) [reply settings/js/0.js]: The file? I think it's because webpack now build two bundles, but I'm not sure.

**Review by rullzer — APPROVED** [review]
Works as expected here :)

**Comment by ChristophWurst** [comment]
@MorrisJobke anything I need to change? Could you have another look, please? :)

**Review by MorrisJobke — APPROVED** [review]
Tested and works really nice 👍  Also together with https://github.com/ChristophWurst/twofactor_admin

**Comment by ChristophWurst** [comment]
Thanks, @MorrisJobke. Should be good now :+1:

**Comment by go2sh** [comment]
Hm, wouldn't it be better to add the 2fa settings to the settings vue app [nextcloud/server/blob/master/settings/src/router.js](router).

**Comment by go2sh** [comment]
Ah, I see. The rest of the encryption tab is not migrated.

**Comment by ChristophWurst** [comment]
> Hm, wouldn't it be better to add the 2fa settings to the settings vue app

No, we're not yet there. Those are the new settings pages that are *completely* rendered with Vue. On the security/encryption page it's mixed :wink:

**Comment by ChristophWurst** [comment]
Looks like one acceptance test run timed out (log just stop) and the other failure is unrelated from what I can tell.

**Pull request merged**